runtime/gap8/gap8.h

```diff
@@ -0,0 +1,31 @@
+#ifndef SHINE_GAP8_H
+#define SHINE_GAP8_H
+
+#include "pmsis.h"
+#include "gaplib/ImgIO.h"
+
+struct ContextImpl {
+    struct pi_device cl_device;
+    struct pi_cluster_conf cl_configuration;
+};
+
+struct KernelImpl {
+    struct pi_cluster_task* cl_task;
+};
+
+struct BufferImpl {
+    void* inner;
+    size_t byte_size;
+};
+
+
+typedef void* DeviceBuffer;
+
+Context createContext(int device_id);
+Kernel loadKernel(void (*handler)(void*), size_t stack_size);
+void launchKernel(
+  Context ctx, Kernel k,
+  int num_threads,
+  size_t arg_count, const KernelArg* args);
+
+#endif
```

Comment:
Maybe this file could be named `gap8/runtime.h`, and we would have the generic `runtime.h` and then `ocl/runtime.h`, etc.

Reply:
I would personally prefer to additionally differentiate those files by naming them differently, just to avoid any potential confusion, but it's a matter of preference I guess

Reply:
We could also consider to pull it out of the folder as in `runtime/gap8.h`?

Reply:
I've left it like this for now. We can rework this later if needed